File: backend/app/models/employee.py
from typing import Literal

from pydantic import BaseModel, Field


EMPLOYEE_LEVELS = ("senior_director", "director", "manager", "ic")
EmployeeLevel = Literal["senior_director", "director", "manager", "ic"]


class Employee(BaseModel):
    id: str = Field(examples=["emp_101"])
    name: str
    department: str
    level: EmployeeLevel
    manager_id: str | None = None

File: frontend/src/types/employee.ts
export type EmployeeLevel = "senior_director" | "director" | "manager" | "ic";

export interface Employee {
  id: string;
  name: string;
  level: EmployeeLevel;
  manager_id: string | null;
}

export const LEVEL_LABELS: Record<EmployeeLevel, string> = {
  senior_director: "Senior Director",
  director: "Director",
  manager: "Manager",
  ic: "Individual Contributor",
};
